### ISSUE TITLE
Add citext table column type

### ISSUE THREAD

**Issue by RodolfoSilva** [hasura/graphql-engine#2038]
Imagine we have the following table:
```sql
CREATE TABLE users (  
  id     uuid PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
  email  text
);
```

If we want to ensure that each user has a unique email we simply add:
```sql
ALTER TABLE users ADD CONSTRAINT email_unique UNIQUE (email);
```
If we try it out by inserting some data:

```sql
INSERT INTO users (email) VALUES ('test@example.com');  
INSERT INTO users (email) VALUES ('test@example.com');

ERROR:  duplicate key value violates unique constraint "email_unique"  
DETAIL:  Key (email)=(test@example.com) already exists. 
```

But if we try insert the same email with upper case:

```sql
INSERT INTO users (email) VALUES ('TEST@example.com');

SELECT * from users;

                  id                  |      email       
--------------------------------------+------------------
 de624d84-63a3-11e9-a923-1681be663d3e | test@example.com
 0c4b25af-5736-4985-9b4c-bdfebd04694d | TEST@example.com
```

I think this can be part of the hasura console.

cc\ @0x777 @dsandip @wawhal

**Comment by rikinsk** [comment]
@RodolfoSilva You can enable using citext by executing `create extension citext;` in the RunSQL section to enable using citext. Showing citext in the column type list  (if the extension is installed) while creating tables will be added in #1933. 

Is there anything else you would expect?

**Comment by RodolfoSilva** [comment]
@rikinsk I think the #1933 Will help me. But why don't enable citext by default?

**Comment by rikinsk** [comment]
@RodolfoSilva Hasura doesn't create anything on a database that it does not depend upon. 

The idea is that Hasura is a layer on top of a database. The database is completely managed by the user. So installing extensions is something better left to the DB admins to decide based on requirements.

**Comment by RodolfoSilva** [comment]
Thanks @rikinsk, my only concern was about making a change to the table column with SQL and this is reflected in the console... I think the issue #1933 will solve this issue :stuck_out_tongue_winking_eye:

**Comment by rikinsk** [comment]
closed by #1933
